scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -90,13 +90,15 @@ def record(self, harness):
                 for instance in harness.recording:
                     cw.writerow(instance)
 
-    def terminate(self, proc):
+    @staticmethod
+    def terminate(proc_pid):
         # encapsulate terminate functionality so we do it consistently where ever
         # we might want to terminate the proc.  We need try_kill_process_by_pid
         # because of both how newer ninja (1.6.0 or greater) and .NET / renode
         # work.  Newer ninja's don't seem to pass SIGTERM down to the children
         # so we need to use try_kill_process_by_pid.
-        for child in psutil.Process(proc.pid).children(recursive=True):
+        proc = psutil.Process(proc_pid)
+        for child in proc.children(recursive=True):
```

Comment:
what you changed, actually it is the same as the old terminate function, so I guess what you need is to replace all process.kill() or process.terminate() by this self.terminate(process)

Reply:
Yes, changed this function a little to fit for other handlers.